Tests/DiligentCoreAPITest/src/MeshShaderTest.cpp

```diff
@@ -0,0 +1,430 @@
+void MeshShaderDrawReferenceVk(ISwapChain* pSwapChain);
+void MeshShaderIndirectDrawReferenceVk(ISwapChain* pSwapChain);
+void AmplificationShaderDrawReferenceVk(ISwapChain* pSwapChain);
+#endif
+
+} // namespace Testing
+
+} // namespace Diligent
+
+using namespace Diligent;
+using namespace Diligent::Testing;
+
+namespace
+{
+
+TEST(MeshShaderTest, DrawQuad)
+{
+    auto* pEnv    = TestingEnvironment::GetInstance();
+    auto* pDevice = pEnv->GetDevice();
+    if (!pDevice->GetDeviceCaps().Features.MeshShaders)
+    {
+        GTEST_SKIP() << "Mesh shader is not supported by this device";
+    }
+
+    auto* pSwapChain = pEnv->GetSwapChain();
+    auto* pConext    = pEnv->GetDeviceContext();
+
+    RefCntAutoPtr<ITestingSwapChain> pTestingSwapChain(pSwapChain, IID_TestingSwapChain);
+    if (pTestingSwapChain)
+    {
+        pConext->Flush();
+        pConext->InvalidateState();
+
+        auto deviceType = pDevice->GetDeviceCaps().DevType;
+        switch (deviceType)
+        {
+#if D3D12_SUPPORTED
+            case RENDER_DEVICE_TYPE_D3D12:
+                MeshShaderDrawReferenceD3D12(pSwapChain);
+                break;
+#endif
+
+#if VULKAN_SUPPORTED
+            case RENDER_DEVICE_TYPE_VULKAN:
+                MeshShaderDrawReferenceVk(pSwapChain);
+                break;
+#endif
+
+            default:
+                LOG_ERROR_AND_THROW("Unsupported device type");
+        }
+
+        pTestingSwapChain->TakeSnapshot();
+    }
+    TestingEnvironment::ScopedReleaseResources EnvironmentAutoReset;
+
+    auto* pContext = pEnv->GetDeviceContext();
+
+    ITextureView* pRTVs[] = {pSwapChain->GetCurrentBackBufferRTV()};
+    pContext->SetRenderTargets(1, pRTVs, nullptr, RESOURCE_STATE_TRANSITION_MODE_TRANSITION);
+
+    float ClearColor[] = {0.f, 0.f, 0.f, 0.f};
+    pContext->ClearRenderTarget(pRTVs[0], ClearColor, RESOURCE_STATE_TRANSITION_MODE_TRANSITION);
+
+    PipelineStateCreateInfo PSOCreateInfo;
+    PipelineStateDesc&      PSODesc = PSOCreateInfo.PSODesc;
+
+    PSODesc.Name = "Mesh shader test";
+
+    PSODesc.PipelineType                                          = PIPELINE_TYPE_MESH;
+    PSODesc.GraphicsPipeline.NumRenderTargets                     = 1;
+    PSODesc.GraphicsPipeline.RTVFormats[0]                        = pSwapChain->GetDesc().ColorBufferFormat;
+    PSODesc.GraphicsPipeline.PrimitiveTopology                    = PRIMITIVE_TOPOLOGY_UNDEFINED; // unused
+    PSODesc.GraphicsPipeline.RasterizerDesc.CullMode              = CULL_MODE_BACK;
+    PSODesc.GraphicsPipeline.RasterizerDesc.FillMode              = FILL_MODE_SOLID;
+    PSODesc.GraphicsPipeline.RasterizerDesc.FrontCounterClockwise = False;
+
+    PSODesc.GraphicsPipeline.DepthStencilDesc.DepthEnable = False;
+
+    const bool IsVulkan = pDevice->GetDeviceCaps().DevType == RENDER_DEVICE_TYPE_VULKAN;
+
+    ShaderCreateInfo ShaderCI;
+    ShaderCI.SourceLanguage             = IsVulkan ? SHADER_SOURCE_LANGUAGE_GLSL : SHADER_SOURCE_LANGUAGE_HLSL;
```

Comment:
This should be SHADER_SOURCE_LANGUAGE_GLSL_VERBATIM (I fixed this), but I don't understand how the test passed for you.

Reply:
I don't test mesh shader on vulkan after changing HLSL to GLSL for vulkan.